src/components/common/ConnectWallet/WalletDetails.tsx

```diff
@@ -1,13 +1,23 @@
-import { Button, Typography } from '@mui/material'
+import { Button, Typography, useMediaQuery } from '@mui/material'
 import type { ReactElement } from 'react'
-
+import { useTheme } from '@mui/material/styles'
 import useOnboard, { connectWallet } from '@/hooks/wallets/useOnboard'
 import { OVERVIEW_EVENTS } from '@/services/analytics/events/overview'
 import KeyholeIcon from '@/components/common/icons/KeyholeIcon'
 import { trackEvent } from '@/services/analytics'
+import { WalletNames } from '@/utils/wallets'
 
 const WalletDetails = ({ onConnect }: { onConnect?: () => void }): ReactElement => {
   const onboard = useOnboard()
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
```

Comment:
Larger tablets won't be deemed as a mobile device. I'd recommend we use `@web3-onboard`'s `getDevice` function instead.

Reply:
It cannot be imported, AFAICT. I added a simple user agent regex.

Reply:
`getDevice` is indeed exported from a declaration file. You [can use `bowser` directly](https://github.com/safe-global/web-core/blob/dev/src/services/pairing/connector.ts#L32-L34) but the regex seems sufficient for now.

It's somewhat out of scope but we can disable hardware wallets directly, e.g. [Trezor](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/trezor/src/index.ts#L26) and [Ledger](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/ledger/src/index.ts#L116). We could have a standard `filter` we add to all apart from WalletConnect.